### ISSUE TITLE
Interactive image classifier - Golem Network Beta.2 bounty

### ISSUE THREAD

**Issue by mat7ias** [golemfactory/yagna#1456]
[Golem Network](http://golem.network/) is a cloud computing power service where everyone can develop, manage and execute workloads in an unstoppable, inexpensive and censorship-free environment.

Since Beta.2 Golem supports a new model of computation – services. In contrast with batch tasks such as what developers were initially limited to in the Beta.1, services are expected to be long-running processes that don't have any natural completion point but rather are started and stopped on explicit command. The goal of this project is to build a service on Golem which enables interaction with an image classifier. The service is expected to allow its users to train the classifier on-line as well as submit images for classification. Details of the classification algorithm are not specified and can be decided by the implementer of the project.

**Requirements**
- The service should be compatible with yagna v0.7.1.
- Starting the service should deploy an interactive image classifier ready to be used.
  - The classifier should be pre-trained for image recognition so that it can be used without excessive further training.
- The service should respond to two kinds of requests:
  - "train on provided data" – includes an image and a label (or a batch of images with labels), no output expected;
  - "classify an image" – includes an image, expected to output a label.
- The classifier should have a decent accuracy when trained on sufficient amount of data.

**Non-requirements**
- The implementer doesn't have to build the model from scratch. Using publicly available pre-trained models is allowed.
- The classifier doesn't have to allow multi-label classification, single-label is enough.
- No scalability is required, the implementer might put reasonable constraints on image size.
- No graphical user interface is required (although it is welcome).
- No direct communication between the user and the provider is required. It is assumed that all communication with the service will be mediated by the requestor application.

**Deliverables**
- A GitHub repository with the following:
  - The service code,
  - A testing script (with training set) allowing to check if the service works as intended,
  - Basic usage instructions,
  - GPL license.
- A video recording demonstrating the usage of the service.

**Resources**
- [Service Model development](https://handbook.golem.network/requestor-tutorials/service-development) – an extensive tutorial on Golem services.
- [yapapi-service-manager](https://github.com/golemfactory/yapapi-service-manager) – a high-level Python library for developing services.

**Estimated time to allocate:** 48 hours

Useful Links:
Bounties Blogpost (including things you need to know!): https://blog.golemproject.net/golem-network-beta-2-bounties/
Beta.2 Blogpost: https://blog.golemproject.net/beta-ii-release/
Docs: https://handbook.golem.network
Install video: https://www.youtube.com/watch?v=Wqm7j7CtQwM
In case you need support, we’re here for you, join our Discord: https://chat.golem.network
Golem Twitter - https://twitter.com/golemproject

**Comment by gitcoinbot** [comment]
Issue Status: 1. **Open** 2. Started 3. Submitted 4. Done 

<hr>

__This issue now has a funding of 11999.9815 GLM (3055.29 USD @ $0.25/GLM)  attached to it.__

 * If you would like to work on this issue you can 'start work' [on the Gitcoin Issue Details page](https://gitcoin.co/issue/golemfactory/yagna/1456/100026046).
* Want to chip in? Add your own contribution [here](https://gitcoin.co/issue/golemfactory/yagna/1456/100026046).
* Questions? Checkout <a href='https://gitcoin.co/help'>Gitcoin Help</a> or the <a href='https://discord.gg/gitcoin/'>Gitcoin's Discord</a>
* $1,938,977.71 more funded OSS Work available on the [Gitcoin Issue Explorer](https://gitcoin.co/explorer)

**Comment by gitcoinbot** [comment]
Issue Status: 1. **Open** 2. Started 3. Submitted 4. Done 

<hr>

__Workers have applied to start work__.


These users each claimed they can complete the work by 265 years, 4 months from now.
Please review their action plans below:


**1) [skotre](https://gitcoin.co/skotre) has applied to start work _(Funders only: [approve worker](https://gitcoin.co/issue/golemfactory/yagna/1456/100026046?mutate_worker_action=approve&worker=skotre) | [reject worker](https://gitcoin.co/issue/golemfactory/yagna/1456/100026046?mutate_worker_action=reject&worker=skotre))_.**

I am interested in working on this, although I don't have any experience regarding this. I don't plan to reinvent the wheel but Golem seems to have a pretty complicated way of doing stuff. Are there any docker/Golem project's that we could use to our advantage?

Learn more [on the Gitcoin Issue Details page](https://gitcoin.co/issue/golemfactory/yagna/1456/100026046).

**Comment by jamessandy** [comment]
This bounty is very interesting and something I can achieve in the shortest possible time, is someone working on it already?

**Comment by gitcoinbot** [comment]
@controlcpluscontrolv Hello from Gitcoin Core - are you still working on this issue? Please submit a WIP PR or comment back within the next 3 days or you will be removed from this ticket and it will be returned to an ‘Open’ status. Please let us know if you have questions!
* [x] reminder (3 days)
* [ ] escalation to mods (6 days)

Funders only: Snooze warnings for [1 day](https://gitcoin.co/issue/golemfactory/yagna/1456/100026046?snooze=1) | [3 days](https://gitcoin.co/issue/golemfactory/yagna/1456/100026046?snooze=3) | [5 days](https://gitcoin.co/issue/golemfactory/yagna/1456/100026046?snooze=5) | [10 days](https://gitcoin.co/issue/golemfactory/yagna/1456/100026046?snooze=10) | [100 days](https://gitcoin.co/issue/golemfactory/yagna/1456/100026046?snooze=100)

**Comment by ControlCplusControlV** [comment]
I am still working on it and in contact with the creator on how to proceed!

**Comment by UtopiusKeller** [comment]
Hello - I just tried to express interest in this project but seems to be taken unavailable.

Is this project still open for development?

Jeff

**Comment by UtopiusKeller** [comment]
What is the status of this bounty please?

**Comment by ControlCplusControlV** [comment]
I am currently working on this bounty, making some changes based off feedback as I get closer to submitting

**Comment by the-emerald** [comment]
Is this still available? It appears that Gitcoin still shows zero applicants. @mariapaulafn @mat7ias

**Comment by CalebEverett** [comment]
I would also be interested in working on this.

**Comment by mariapaulafn** [comment]
hi @CalebEverett @the-emerald as far as concerned, the bounty is still being worked on by @ControlCplusControlV 
if its abandoned i will let you know

**Comment by CalebEverett** [comment]
@mariapaulafn Thanks - would it be at all useful to submit an alternate version even if not for the bounty - is there some variation that would be more useful? I was primarily interested in doing this for the learning opportunity.

**Comment by xiaods** [comment]
Is this bounty still open ? I'd love to help with it

**Comment by yashpatel5400** [comment]
Hey! I'm just gonna parrot all the comments above and ask if this bounty is still open. Would be very interested in exploring this if so!

**Comment by ControlCplusControlV** [comment]
I am working on this bounty, you can view my progress on this repo https://github.com/ControlCplusControlV/Golem-Image-Classifier/tree/coderev2

**Comment by aahutsal** [comment]
@mat7ias think, you should close both: bounty and issue
